### PR TITLE
Fix inheritance.sub.html WPT to work on Servo

### LINKED ISSUE #31518: Fix inheritance.sub.html WPT to work on Servo
https://wpt.fyi/results/css/css-backgrounds/inheritance.sub.html?label=master&label=experimental&product=chrome&product=firefox&product=safari&product=servo&aligned

There are some subtests that are broken on Servo, e.g. "Property border-top-width has initial value 3px" becomes "Property border-top-width has initial value undefined" and fails.

The reason is that the test uses `mediumWidth` as the initial value, which is defined as https://github.com/servo/servo/blob/3a3e76a935f92ce84c24496cfc46207cd46088f6/tests/wpt/tests/css/css-backgrounds/inheritance.sub.html#L34

Servo doesn't support `column-rule-width`, so that's undefined. But referring to `column-rule-width` is weird in the 1st place, and https://github.com/w3c/csswg-drafts/commit/075522eb7269273709eda0037c8848879f87ac02 simply defined `medium` as `3px`.

So the test could be fixed to just use `3px` as the initial border width value, and then update the expectations for both the new and the legacy layout engines.

See https://github.com/servo/servo/blob/main/tests/wpt/README.md for details about running WPT and updating the expectations.

### PR DESCRIPTION
**Pull request by azharcodeit**
Fix inheritance.sub.html WPT to work on Servo

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #31518 (GitHub issue number if applicable)

- [x] There are tests for these changes

### CODE REVIEW

**Review by Loirooriol — CHANGES_REQUESTED** [review]
Thanks, but while you are at it you should purge the old entries from the expectation files, like

```
rm tests/wpt/meta/css/css-backgrounds/inheritance.sub.html.ini
./mach test-wpt -r --log-raw /tmp/servo.log --headless tests/wpt/tests/css/css-backgrounds/inheritance.sub.html
./mach update-wpt /tmp/servo.log

rm tests/wpt/meta-legacy-layout/css/css-backgrounds/inheritance.sub.html.ini
./mach test-wpt --legacy-layout -r --log-raw /tmp/servo.log --headless tests/wpt/tests/css/css-backgrounds/inheritance.sub.html
./mach update-wpt --legacy-layout /tmp/servo.log
```

(Typically you don't need to remove to update the expectations, but in this case it's for purging the obsolete entries)

**Comment by Loirooriol** [comment]
Oh, and when you post a pull request, you should mark the checkboxes in the template, like

```
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #31518 (GitHub issue number if applicable)
- [X] These changes do not require tests because this is just fixing a test
```

(arguably fixing a test could also count as "There are tests for these changes", pick whatever you think makes more sense)

**Comment by Loirooriol** [comment]
You added an empty patch, possibly by mistake and you wanted to push another one?

**Comment by azharcodeit** [comment]
> You added an empty patch, possibly by mistake and you wanted to push another one?

Yes, working on it, pulled main branch so must have been automatically updated

File: tests/wpt/tests/css/css-backgrounds/inheritance.sub.html [hunk 1]
```diff
@@ -31,7 +31,7 @@
 </style>
 <script>
 const transparentColor = 'rgba(0, 0, 0, 0)'; // https://www.w3.org/TR/css-color-3/#transparent
-const mediumWidth = getComputedStyle(document.getElementById('reference')).columnRuleWidth; // e.g. 3px
+const mediumWidth = '3px';
```
[the remark this answers is not in the document]
Comment (azharcodeit) [reply tests/wpt/tests/css/css-backgrounds/inheritance.sub.html]: @Loirooriol, addressed fixes in updated PR, please have a look, thank you!

**Review by Loirooriol — APPROVED** [review]
Thanks, looks good!

BTW, it's noteworthy that since this is a WPT change, it has been exported to that repo: https://github.com/web-platform-tests/wpt/pull/44954
There you can go to the "Checks" tab to ensure that this isn't breaking on other browsers. Effectively, chrome, firefox and safari were passing 64/64 before the change, and still 64/64 after the change, so everything is good.

**Comment by azharcodeit** [comment]
> Thanks, looks good!
> 
> BTW, it's noteworthy that since this is a WPT change, it has been exported to that repo: [web-platform-tests/wpt#44954](https://github.com/web-platform-tests/wpt/pull/44954) There you can go to the "Checks" tab to ensure that this isn't breaking on other browsers. Effectively, chrome, firefox and safari were passing 64/64 before the change, and still 64/64 after the change, so everything is good.

Great! I will check it, thank you

**Comment by Loirooriol** [comment]
Ah, #31453 landed first, so now there is a conflict in `tests/wpt/meta/css/css-backgrounds/inheritance.sub.html.ini`
Can you rebase the patch to the current tip?

File: tests/wpt/meta/css/css-backgrounds/inheritance.sub.html.ini [hunk 2]
```diff
@@ -1,12 +1,6 @@
...
-  [Property border-left-width has initial value undefined]
-    expected: FAIL
-
-  [Property border-right-width has initial value undefined]
-    expected: FAIL
-
-  [Property border-top-width has initial value undefined]
-    expected: FAIL
+  [Property box-shadow does not inherit]
+    expected: FAIL
```
Comment (Loirooriol) [inline tests/wpt/meta/css/css-backgrounds/inheritance.sub.html.ini:6]: You shouldn't be adding this. You can just remove this file now.

**Comment by sagudev** [comment]
> 🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

https://github.com/web-platform-tests/wpt/pull/44954 does not include comment with spec link, so it's really out of sync.

**Comment by Loirooriol** [comment]
It's strange, because initially the bot could export the changes without problems, and the test hasn't been changed upstream: https://github.com/web-platform-tests/wpt/commits/master/css/css-backgrounds/inheritance.sub.html

I will investigate tomorrow.

**Comment by Loirooriol** [comment]
I squashed the commits together and now it works.

I suspect the problem was that d5908c9ba80aff1f81c67519ed0f8b2c9d454064 has no WPT modification, and https://github.com/servo/servo/blob/ef3dad3a610d9d73754e976bece64a2184ed3821/python/wpt/exporter/step.py#L156 isn't using `--allow-empty`.

**Comment by mrobinson** [comment]
Looks like the failure here was the typical "hdiutil: create failed - Resource busy" that we see.

**Pull request merged**